proto/query.proto

```diff
@@ -883,6 +883,9 @@ message RealtimeStats {
 
   // view_schema_changed is to provide list of views that have schema changes detected by the tablet.
   repeated string view_schema_changed = 8;
+
+  // udfs_changed is used to signal that the UDFs have changed on the tablet.
+  bool udfs_changed = 9;
```

Comment:
For tables and views, the signal that a change has occurred is sent as the name of the table/view. Vtgate can then fetch the new definition for these objects when it wants that info. We did this so that the healtstream doesn't get weighed down with extra traffic that the vtgate might not be interested in - it can decide when to fetch the heavy data load.

For UDFs, the data load is the list and types of UDFs. The vtgate will simply reload the full list when something has changed. With this model, there is no need for the signal to specify what has changed.